libhfst/src/HfstTransducer.cc

```diff
@@ -3113,13 +3068,13 @@ void encode_flag_diacritics(HfstTransducer &fst)
 
   // copy alphabet, encode all flags
   StringSet alpha = basic_fst.get_alphabet();
-  for (StringSet::const_iterator it = alpha.begin(); it != alpha.end(); it++)
+  for (const auto & it : alpha)
     {
-      if (it->size() > 4)
+      if (it.size() > 4)
         {
-          if ((it->at(0) == '%') && (it->at(it->size()-1) == '%'))
+          if ((it.at(0) == '%') && (it.at(it.size()-1) == '%'))
```

Comment:
Hm, there's `.at()` in HFST? Any use of `.at()` should be questioned. Usually `op[]` is fine, and it certainly is here.

Reply:
Ok, fixed.